### PR TITLE
Add doc on managing environments

### LINKED ISSUE #6702: Documentation: how to Manage Python Dependencies
I feel like this is something that should be covered in the documentation for Dask, but couldn't really find it. Basically, how do people make sure python code in other project files is accessible to the workers?
https://stackoverflow.com/questions/63944610/how-do-people-generally-manage-dependencies-for-their-tasks-in-dask

### PR DESCRIPTION
**Pull request by martindurant**
- [ ] ~Tests added / passed~
- [ ] ~Passes `black dask` / `flake8 dask`~

Closes https://github.com/dask/distributed/pull/4190
Fixes #6702

### CODE REVIEW

File: docs/source/setup/environment.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+simply shared rather than serialised and deserialised.
+
+Similarly, if you use the multiprocessing scheduler, new processes
+will be copied from, or launched in the same way as the original process,
+so you only need make sure that you have not changed environment variables
+related to starting up python and importing
+code (such as PATH, PYTHONPATH, ``sys.path``).
+
+If you are using the distributed scheduler on a single machine, this is roughly
+equivalent to using the multiprocessing scgheduler, above, if you are launching
```
Comment (jsignell) [inline docs/source/setup/environment.rst:32]: typo
```suggestion
equivalent to using the multiprocessing scheduler, above, if you are launching
```

File: docs/source/setup/environment.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+Maintain consistent environments
+````````````````````````````````
+
+If you manage your environments yourself, then setting up module consistency
+can be as simple as creating environments from the same pip or conda specification
+on each machine. You should consult the documentation for ``pip``, ``pipenv``
+and ``conda``, whichever you normally use. You will normally want to be as specific
+about package versions as possible, and distribute the same environment file to
+workers before installation.
+
```
Comment (jsignell) [inline docs/source/setup/environment.rst:49]: Maybe should mention the `preload` argument here?

**Review by jsignell — COMMENTED** [review]
I like the idea of this very much, just a few suggestions.

File: docs/source/setup/environment.rst (hunk 2 above)
Comment (martindurant) [reply docs/source/setup/environment.rst:49]: Yeah, wasn't sure - should we have an "expert" section?

Comment (jsignell) [reply docs/source/setup/environment.rst:49]: That sounds good.

File: docs/source/setup/environment.rst [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  many requests
+- cluster install method (e.g., `parcels`_): depending on your infrastructure, there may be
+  ways to install specific binaries to all workers in a cluster.
+
+.. _conda-pack: https://conda.github.io/conda-pack/
+.. _parcels: https://docs.cloudera.com/documentation/enterprise/latest/topics/cm_ig_parcels.html
+
+Temporary installations
+```````````````````````
+The worker plugin ``distributed.diagnostics.plugin.PipInstall`` allows you to
```
Comment (TomAugspurger) [inline docs/source/setup/environment.rst:70]: I think :class:`distributed.diagnostics.plugin.PipInstall` will work if you add

```
    "distributed": (
        "https://distributed.dask.org/en/latest/",
        "https://distributed.dask.org/en/latest/objects.inv",
    ),
```

to `docs/source/conf.py`.

Comment (martindurant) [reply docs/source/setup/environment.rst:70]: It works!

File: docs/source/setup/environment.rst (hunk 2 above)
Comment (martindurant) [reply docs/source/setup/environment.rst:49]: I put it in the following block instead, since the list here is for less ephemeral state

**Comment by martindurant** [comment]
@jsignell ?

**Review by TomAugspurger — APPROVED** [review]

**Pull request merged**